pyo3-macros-backend/src/utils.rs

```diff
@@ -144,11 +144,41 @@ pub fn unwrap_ty_group(mut ty: &syn::Type) -> &syn::Type {
     ty
 }
 
-/// Extract the path to the pyo3 crate, or use the default (`::pyo3`).
-pub(crate) fn get_pyo3_crate(attr: &Option<CrateAttribute>) -> syn::Path {
-    match attr {
-        Some(attr) => attr.value.0.clone(),
-        None => syn::parse_str("::pyo3").unwrap(),
+pub struct Ctx {
+    pub pyo3_path: PyO3CratePath,
+}
+
+impl Ctx {
+    pub(crate) fn new(attr: &Option<CrateAttribute>) -> Self {
+        let pyo3_path = match attr {
+            Some(attr) => PyO3CratePath::Given(attr.value.0.clone()),
+            None => PyO3CratePath::Default,
+        };
+
+        Self { pyo3_path }
+    }
+}
+
+pub enum PyO3CratePath {
+    Given(syn::Path),
+    Default,
+}
+
+impl PyO3CratePath {
+    pub fn to_tokens_spanned(&self, span: Span) -> TokenStream {
+        match self {
+            Self::Given(path) => quote::quote_spanned! { span => #path },
+            Self::Default => quote::quote_spanned! {  span => ::pyo3 },
+        }
+    }
+}
```

Comment:
The idea of adding this here is that this gave back control to the span that the `::pyo3` default path was being emitted with. 

I applied in manually in the places where it mattered. If we wanted to force ourselves to think about it we could remove `quote::ToTokens` implementation and require ourselves to always call this method. I quite like the convenience of the `ToTokens` implementation though :)